lib/media/track/localtrackpublication.js

```diff
@@ -48,38 +39,47 @@ class LocalTrackPublication {
       track: {
         enumerable: true,
         value: track
-      },
-      trackName: {
-        enumerable: true,
-        value: track.name
-      },
-      trackSid: {
-        enumerable: true,
-        value: sid
       }
     });
+
+    track.on('disabled', this._reemitTrackEvent);
+    track.on('enabled', this._reemitTrackEvent);
   }
 
   toString() {
     return `[LocalTrackPublication #${this._instanceId}: ${this.trackSid}]`;
   }
 
+  /**
+   * Whether the published {@link Track} is enabled
+   * @property {boolean}
+   */
+  get isTrackEnabled() {
+    return this.track.kind === 'data' ? true : this.track.isEnabled;
+  }
+
   /**
    * Unpublish a {@link LocalTrackPublication}. This means that the media
    * from this {@link LocalTrackPublication} is no longer available to the
    * {@link Room}'s {@link RemoteParticipant}s.
    * @returns {this}
    */
   unpublish() {
+    this.track.removeListener('disabled', this._reemitTrackEvent);
+    this.track.removeListener('enabled', this._reemitTrackEvent);
     this._unpublish(this);
     return this;
   }
 }
 
 /**
- * {@link LocalTrackPublication} options
- * @typedef {object} LocalTrackPublicationOptions
- * @property {LogLevel|LogLevels} logLevel - Log level for 'media' modules
+ * The published {@link Track} was disabled.
+ * @event TrackPublication#trackDisabled
+ */
+
+/**
+ * The published {@link Track} was enabled.
+ * @event TrackPublication#trackEnabled
  */
```

Comment:
These declarations should eventually move to trackpublication.js. We can do that now or later.